articles/node_lifecycle.md

```diff
@@ -196,51 +199,176 @@ Transitions to `ErrorProcessing` may be caused by error return codes in callback
+
 ### Destroy Transition
 
 This transition will simply cause the deallocation of the node.
 In an object oriented environment it may just involve invoking the destructor.
 Otherwise it will invoke a standard deallocation method.
 This transition should always succeed.
 
-### Create Transition
+## Management Interface
 
-This transition will instantiate the node, but will not run any code beyond the constructor.
+A node that has a managed life cycle complying with the above life cycle shall provide the following interface via ROS topics and services.
+This interface is to be used by tools to manage the node's life cycle state transitions, either automatically or manually according to the tool's purpose.
 
-## Management Interface
+The topics and services of this interface shall function in all states of the node's life cycle.
+They shall not be disabled by the node shifting to the `Inactive` state, for example.
 
-A managed node will be exposed to the ROS ecosystem by the following interface, as seen by tools that perform the managing.
-This interface should not be subject to the restrictions on communications imposed by the lifecycle states.
+### Interface namespace
 
-It is expected that a common pattern will be to have a container class which loads a managed node implementation from a library and through a plugin architecture automatically exposes the required management interface via methods and the container is not subject to the lifecycle management.
-However, it is fully valid to consider any implementation which provides this interface and follows the lifecycle policies a managed node.
-Conversely, any object that provides these services but does not behave in the way defined in the life cycle state machine is malformed.
+The interface shall be provided in a namespace named "infra/lifecycle" underneath the node's namespace.
+
+For example, if a node named `talker` has a managed life cycle complying with the state machine described above, it shall provide topics under the namespace `/talker/infra/lifecycle`.
+
+All examples in the following sections are also given assuming a node named `talker`.
+
+If the `infra/lifecycle` namespace is available under a node's namespace, then that node shall be assumed to be functioning according to the managed life cycle.
+If the node is not functioning according to the managed life cycle, the `infra/lifecycle` namespace shall not exist.
+In other words, tooling shall judge if a node is managed or not by the presence of the `infra/lifecycle` namespace.
+
+### State enumerations
+
+The messages used by the interface shall use the following enumeration for indicating states.
+
+    uint8 UNKNOWN=0
+    uint8 UNCONFIGURED=1
+    uint8 INACTIVE=2
+    uint8 ACTIVE=3
+    uint8 FINALIZED=4
+    uint8 CONFIGURING=10
+    uint8 CLEANING_UP=11
+    uint8 SHUTTING_DOWN=12
+    uint8 ACTIVATING=13
+    uint8 DEACTIVATING=14
+    uint8 ERROR_PROCESSING=15
```

Comment:
I don't mind either way. Linking to the spec might have the problem of tracking changes over time. Then again it may be just as hard to represent that in this document as well. So I don't see that one way is strictly better than the other. At least with the interface file being the sole authority, you can avoid some duplication of information if you choose.